chore(deps): Bump `@metamask/approval-controller` from `^7.0.1` to `^7.1.0`

MajorLift opened the pull request:
## **Description**

```md
### Changed

- Bump `@metamask/approval-controller` from `^7.0.1` to `^7.1.0` ([#11116](https://github.com/MetaMask/metamask-mobile/pull/11116))
```

## **Related issues**

- Closes https://github.com/MetaMask/metamask-mobile/issues/11178

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

MajorLift committed:
Bump `@metamask/approval-controller` from `^7.0.1` to `^7.0.3`

MajorLift committed:
Fix lockfile

MajorLift committed:
Revert "Restore `@ts-expect-error` for base-controller alignment on `ApprovalControllerMessenger`"

This reverts commit bb07086640243bbdee6962b9a1b8773b5c4ca792.

cryptodev-2s reviewed (changes requested):
can we remove also `patches/@metamask+approval-controller+7.0.2.patch` or at least we maintain the patch with the new bumped version ?

MajorLift committed:
Bump `@metamask/base-controller` from `^6.0.0` to `^7.0.0`

MajorLift committed:
Add temporary `ts-expect-error` directives for base-controller version mismatches

- These are because mobile's base-controller version is now ahead of the version used in the controllers
- These will be removed during the controller upgrades

MajorLift committed:
Replace all `ts-ignore` base-controller version mismatch directives with `ts-expect-error`

- There is never a reason to use `ts-ignore` over `ts-expect-error`, as it fails silently when the error it is suppressing is removed.

MajorLift committed:
Add `ts-expect-error` directive for `SelectedNetworkController`

mikesposito committed:
Merge branch 'main' into jongsun/chore/bump/approval-controller-7.0.3

cryptodev-2s committed:
Merge remote-tracking branch 'origin/main' into jongsun/chore/bump/approval-controller-7.0.3

mikesposito committed:
Merge branch 'main' into jongsun/chore/bump/approval-controller-7.0.3

mikesposito committed:
fix: messed up merge commit

mikesposito committed:
fix: remove unused ts-expect-error directive

mikesposito committed:
chore: bump to version 7.1.0

mikesposito committed:
chore: deduplicate packages

mikesposito committed:
remove patch file

mcmire reviewed (commented):
Looks good overall but there are some conflicts now it seems.

cryptodev-2s committed:
Merge remote-tracking branch 'origin/main' into jongsun/chore/bump/approval-controller-7.0.3

cryptodev-2s commented:
closed as completed in https://github.com/MetaMask/metamask-mobile/pull/11911
